ts/Series/Flags/FlagsSeries.ts

```diff
@@ -419,6 +419,10 @@ class FlagsSeries extends ColumnSeries {
                 (plotX as any) >= 0 &&
                 !outsideRight
             ) {
+                // #15384
+                if (graphic && shape && shape !== graphic.symbolKey) {
+                    graphic = point.graphic = graphic.destroy();
+                }
```

Comment:
We have a point method for this, called `hasNewShapeType`. Plus, we don't have to unset `point.graphic` because it is reassigned directly after. So this check can be shortened to:

```ts
                // #15384
                if (graphic && point.hasNewShapeType()) {
                    graphic = graphic.destroy();
                }
```

Reply:
I dont think it can be used for this, not in its current state at least, in it `oldShapeType` is `g`, `this.shapeType` is `rect`, so it always returns true and neither of them have anything to do with `series.options.shape`/`point.options.shape` in this case.

Reply:
We had a similar issue with 3D, where point's graphic is a group: `<g>`, but shapeType is `rect`. 3D overwrites `hasNewShapeType` and adds its own check. If we want to follow the convention, then `point.hasNewShapeType()` should be defined for Flag series too

Reply:
Actually I think we should come up with a general check that applies in all cases. The idea of the `hasNewShapeType` function is to determine when we need to generate a new graphic in cases like casting series type or switching between 2D and 3D. The 3D override seems to be too categoric since it always returns false when the graphic is grouped. Basically the idea is to return false only when the existing type can _not_ be animated into the new type. What do you think @pawelfus ?

Reply:
> Basically the idea is to return false only when the existing type can not be animated into the new type.

We don't have morphism implemented (only [v7 demo](https://jsfiddle.net/BlackLabel/dhrxcs1m/8/)), so at this moment none of the shapes can be animated into the new one, but correct me if I'm wrong. We use `hasNewShapeType` method to check if the new shape was set and if yes, then destroy the old graphic. All the other visual aspects can be animated (color, radius, width etc.).

Regarding the solution, maybe instead of checking the node's name, we can check `point.shapeType`?  In 3d we have `arc3d` and `cuboid`. The flags series inherits `shapeType = rect` from column series. While every flag is a label, we can still use `shapeType`. This should be smart enough and safe too. The only problem I see, is probably with `shapeType` being changed before we draw a shape (`hasNewShapeType()` called too late?).

Reply:
But we still have the flags, where I don't think the shape is reflected in `shapeType`. I'll merge this at the current state, then we can revive the problem when we get to morphing.